src/Lachain.Core/Blockchain/VM/ExternalHandler.cs

```diff
@@ -378,6 +378,53 @@ public static void Handler_Env_GetMsgValue(int dataOffset)
             if (!ret)
                 throw new InvalidContractException("Bad call to (get_msg_value)");
         }
+        
+        public static void Handler_Env_GetExternalBalance(int addressOffset, int resultOffset)
+        {
+            Logger.LogInformation($"Handler_Env_GetExternalBalance({addressOffset}, {resultOffset})");
+            var frame = VirtualMachine.ExecutionFrames.Peek() as WasmExecutionFrame
+                        ?? throw new InvalidOperationException("Cannot call GetExternalBalance outside wasm frame");
+            
+            // Get the address from the given memory offset
+            var snapshot = frame.InvocationContext.Snapshot;
+            var addressBuffer = SafeCopyFromMemory(frame.Memory, addressOffset, 20);
+            if (addressBuffer is null)
+                throw new InvalidContractException("Bad call to call function");
+            var address = addressBuffer.Take(20).ToArray().ToUInt160();
+            
+            // Get balance at the given addres
+            var balance = snapshot.Balances.GetBalance(address);
+            
+            // Load balance at the given resultOffset
+            var result = SafeCopyToMemory(frame.Memory, balance.ToUInt256().ToBytes(), resultOffset);
+
+            if (!result)
+                throw new InvalidContractException("Bad call to (get_external_balance)");
+        }
+        
+        public static void Handler_Env_GetBlockTimestamp(int dataOffset)
+        {
+            Logger.LogInformation($"Handler_Env_GetBlockTimestamp()");
+            var frame = VirtualMachine.ExecutionFrames.Peek() as WasmExecutionFrame
+                        ?? throw new InvalidOperationException("Cannot call GetBlockTimestamp outside wasm frame");
+
+            // Get the TotalBlockHeight at the given Snapshot
+            var snapshot = frame.InvocationContext.Snapshot;
+            var blockHeight = snapshot.Blocks.GetTotalBlockHeight();
+            
+            // Get block at the given height
+            var block = snapshot.Blocks.GetBlockByHeight(blockHeight);
+            
+            // Get block's timestamp
+            if (block is null)
+                throw new InvalidContractException("Bad call to call function");
```

Comment:
I think it is better to have a more specific message here:
```suggestion
                throw new InvalidContractException("Bad call to (get_block_timestamp)");
```

Reply:
incorporated the comments